### PR TITLE
Di 20810 update schema on tsa creation

### PR DESCRIPTION
**Pull request by saargt**

### CODE REVIEW

File: server-side/api.ts [hunk 1]
```diff
@@ -160,7 +161,16 @@ async function searchFunctionAdapter(client: Client, request: Request, resourceN
 }
 // #endregion
 
+export async function handle_tsa_creation(client: Client, request: Request)
+{
```
Comment (ShirMarciano) [inline server-side/api.ts:165]: It is not only on TSA creation - TSA can by update also (the Name can be changed

Comment (ShirMarciano) [reply server-side/api.ts:165]: Also you are missing function return Type, add the lint to you project's :)

File: server-side/installation.ts [hunk 2]
```diff
@@ -272,18 +279,32 @@ async function upsertRelation(papiClient: PapiClient, relation: Relation)
...
-	const missingSchemas: Array<string> = RESOURCE_TYPES.filter(resource => !existingSchemas.includes(resource));
-
-	return missingSchemas;
+	const subscription: Subscription = {
+		AddonUUID: AddonUUID,
+		Name: TSA_CREATION_SUBSCRIPTION_NAME,
+		Type: "data",
+		FilterPolicy: {
+			Action: ['insert'],
+			AddonUUID: [CORE_ADDON_UUID],
```
Comment (ShirMarciano) [inline server-side/installation.ts:302]: You should subscribe also for update

File: server-side/schema.service.ts [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+        return missingSchemas;
+    }
+
+    /**
+     * Add TSA fields to schemas
+     * @param tsaKeys TSA key(s) to get to add to schemas
+     */
+    public async addTsaFieldToSchema(tsaKeys: string | string[]): Promise<void>
+    {
```
Comment (ShirMarciano) [inline server-side/schema.service.ts:86]: Talk with me on there two functions - addTsaFieldToSchema and addTsaFieldsToSchemas

File: server-side/installation.ts (hunk 2 above)
Comment (saargt) [reply server-side/installation.ts:302]: @ShirMarciano I couldn't figure out how you could update an existing TSA. type_safe_attribute endpoint has UUID and InternalID, but not Label nor FieldID. meta_data endpoint has Label and FieldID, but no UUID. Calling meta_data with InternalID in POST body result in an entirley new TSA, the passed InternalID is disregarded.

File: server-side/api.ts [hunk 4]
```diff
@@ -186,10 +186,24 @@ export async function handle_tsa_creation(client: Client, request: Request)
...
+
+export async function handle_tsa_modifications(client: Client, request: Request)
+{
+	console.log('Handling TSA modifications');
+
+	const papiClient = Helper.getPapiClient(client);
+	const tsaService = new TsaService(papiClient);
+
+	const modifiedObjects: { Key: string; OldName: string; }[] = request.body.Message?.ModifiedObjects?.map(modifiedObject => 
+		({ Key: modifiedObject?.ObjectKey, OldName: modifiedObject?.ModifiedFields[0].OldValue }
```
Comment (ShirMarciano) [inline server-side/api.ts:203]: OldValue - please do not use this value

File: server-side/tsa-service/tsa.service.ts [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    * Add TSA fields to schemas
+    * @param tsas TSA fields to add to schemas
+    * @param schemas Schemas to add TSA fields to
+    * @private
+    */
+    private addTsaFieldsToSchemasFieldsObject(tsas: TSA[],
+                                schemas: SchemaNameAndFields[]): void
+    {
+        console.log(`Adding TSA fields to schemas: ${tsas.map(tsa => tsa.Name).join(', ')}`);
+        console.log(`Schemas: ${schemas.map(schema => schema.Name).join(', ')}`);
```
Comment (ShirMarciano) [inline server-side/tsa-service/tsa.service.ts:182]: if you will create schemaNameToFieldsMap before the logs you can for the log do 
schemaNameToFieldsMap.keys()

File: server-side/tsa-service/tsa.service.ts [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+            let schemaFields: { [key: string]: SchemeField } | undefined; 
+            if(OwnerObjectTypeIDToResourceTypeMap.has(tsa.OwnerObjectTypeID))
+            {
+                schemaFields = schemaNameToFieldsMap.get(OwnerObjectTypeIDToResourceTypeMap.get(tsa.OwnerObjectTypeID)!);
+            }
+
+            if (schemaFields) 
+            {
+                schemaFields[tsa.Name] = {
+                    Type: getAdalFieldTypeFromPopulatableObjectType(tsa.PopulatableObjectType),
```
Comment (ShirMarciano) [inline server-side/tsa-service/tsa.service.ts:199]: where is the schemaFields inserted to the schema?

File: server-side/tsa-service/tsa.service.ts [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+    }
+
+    /**
+    * Add TSA fields to schemas
+    * @param tsas TSA fields to add to schemas
+    * @param schemas Schemas to add TSA fields to
+    * @private
+    */
+    private addTsaFieldsToSchemasFieldsObject(tsas: TSA[],
+                                schemas: SchemaNameAndFields[]): void
```
Comment (ShirMarciano) [inline server-side/tsa-service/tsa.service.ts:179]: Talk with me again on this function

**Pull request merged**